apps/aens/src/aens_claim_tx.erl

```diff
@@ -0,0 +1,210 @@
+%%%=============================================================================
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Module defining the Naming System claim transaction
+%%% @end
+%%%=============================================================================
+
+-module(aens_claim_tx).
+
+-include("ns_txs.hrl").
+-include_lib("apps/aecore/include/common.hrl").
+-include_lib("apps/aecore/include/trees.hrl").
+
+-behavior(aetx).
+
+%% Behavior API
+-export([new/1,
+         fee/1,
+         nonce/1,
+         origin/1,
+         check/3,
+         process/3,
+         signers/1,
+         serialize/1,
+         deserialize/1,
+         type/0,
+         for_client/1
+        ]).
+
+%% Getters
+-export([account/1,
+         name/1]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-define(NAME_CLAIM_TX_TYPE, <<"name_claim">>).
+-define(NAME_CLAIM_TX_VSN, 1).
+
+-opaque claim_tx() :: #ns_claim_tx{}.
+
+-export_type([claim_tx/0]).
+
+%%%===================================================================
+%%% Behaviour API
+%%%===================================================================
+
+-spec new(map()) -> {ok, claim_tx()}.
+new(#{account   := AccountPubKey,
+      nonce     := Nonce,
+      name      := Name,
+      name_salt := NameSalt,
+      fee       := Fee}) ->
+    {ok, #ns_claim_tx{account   = AccountPubKey,
+                      nonce     = Nonce,
+                      name      = Name,
+                      name_salt = NameSalt,
+                      fee       = Fee}}.
+
+-spec fee(claim_tx()) -> integer().
+fee(#ns_claim_tx{fee = Fee}) ->
+    Fee.
+
+-spec nonce(claim_tx()) -> non_neg_integer().
+nonce(#ns_claim_tx{nonce = Nonce}) ->
+    Nonce.
+
+-spec origin(claim_tx()) -> pubkey().
+origin(#ns_claim_tx{account = AccountPubKey}) ->
+    AccountPubKey.
+
+-spec check(claim_tx(), trees(), height()) -> {ok, trees()} | {error, term()}.
+check(#ns_claim_tx{account = AccountPubKey, nonce = Nonce,
+                   fee = Fee, name = Name, name_salt = NameSalt}, Trees, Height) ->
+    %% TODO: Maybe include burned fee in tx fee. To do so, mechanism determining
+    %% which part of fee goes to miner and what gets burned is needed.
+    %% One option is to change tx:fee/1 callback to tx:fee_for_miner/1.
+    BurnedFee = aec_governance:name_claim_burned_fee(),
```

Comment:
From https://github.com/aeternity/protocol/blob/da3a79f9399f73f78833bb74f51ddd66b49ac647/AENS.md#mechanisms :
> Fees are the main mechanism to discourage spam and squatting. This initial version will burn the governance fee, in order to enable us to change this behaviour in the future. If we start out by giving this fee to miners, they will be very hesitant to accept any changes, which will impact their income negatively. Thus starting out with burning could allow us an easier path for future update to the fee structures.

From https://github.com/aeternity/protocol/blob/da3a79f9399f73f78833bb74f51ddd66b49ac647/AENS.md#claim :
> Note that the fee here is a distinct fee from the normal transaction fee.

From the above I had implied that the claim tx was meant to contain two fees: normal fee for miner, and fee-to-be-burned (that https://github.com/aeternity/protocol/blob/da3a79f9399f73f78833bb74f51ddd66b49ac647/AENS.md#claim calls simply as `fee`).

So the comment in the code above appears to be out of sync with protocol doc - at least in my understanding.